code/game/turfs/trench.dm

```diff
@@ -0,0 +1,25 @@
+obj/structure/trench
+	name = "Trench"
+	desc = "A wide, narrow earthwork designed to protect soldiers from any projectiles, since WWI!"
+	icon = 'icons/turf/trench.dmi'
+	icon_state = "gold0"
+	density = FALSE
+	anchored = TRUE
+	var/slowamt = 4
+	tiles_with = list(
+		/obj/structure/trench,
+	)
+
+obj/structure/trench/Initialize()
+	. = ..()
+	relativewall()
+	relativewall_neighbours()
+
+obj/structure/trench/Crossed(atom/movable/AM)
+
+	var/mob/living/H = AM
+	H.next_move_slowdown += slowamt
+	ADD_TRAIT(H, TRAIT_ISINTRENCH, TRAIT_SOURCE_TRENCH)
```

Comment:
Really not convinced by the use of crossed and everything. Imo, you should leap in/out of the trench, you should not be slown when moving from one trench to the other. Here you just walk on the trench and tada, you're in the trench

Reply:
So, you want slowdown when going into a trench but not among trenches?

Reply:
That would seems more plausible? I was also thinking about a windup when going in / out trenches, seems better than a slowdown.

That's probably your choice though

Reply:
I get what you are saying and I agree, it should work like that. The question is if it is possible.